### PR TITLE
Remove keycloak-core and keycloak-crypto-default from SAML galleon fe…

### LINKED ISSUE #32586: Remove keycloak-core and keycloak-crypto-default from SAML galleon feature pack and upgrade them to Java 17
### Description

It looks that `keycloak-core` and `keycloak-crypto-default` can be safely removed from SAML distribution feature pack as they are likely not needed here.

They are declared in the distribution by the SAML galleon feature pack as the dependencies of the Wildfly/EAP module `keycloak-saml-adapter-core`. However they are not needed as dependencies of that module. In the codebase, they are declared just as "test" dependencies of the corresponding module - https://github.com/keycloak/keycloak/blob/main/adapters/saml/core/pom.xml#L67 .

This will also allow to make these modules upgraded to Java 17.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
…ature pack and upgrade them to Java 17

closes #32586

### CODE REVIEW

**Review by jonkoops — DISMISSED** [review]
LGTM

File: core/src/main/java/org/keycloak/jose/jwk/JWKParser.java [hunk 1]
```diff
@@ -48,4 +60,65 @@ public JWKParser parse(String jwk) {
         }
     }
 
+    @Override
+    public PublicKey toPublicKey() {
+        String keyType = jwk.getKeyType();
```
Comment (sschu) [inline core/src/main/java/org/keycloak/jose/jwk/JWKParser.java:65]: I think there is way to create this without jwk (see `static JWKParser create()` above) so maybe better reverse the following checks to not have a NullPointerException.

Comment (mposolda) [reply core/src/main/java/org/keycloak/jose/jwk/JWKParser.java:68]: Good point.

I've just pretty much copy/pasted this part from the original `JWKParser` class in the `java16` directory as there is no need to have specific code for java16 or bigger. But at the same time, it doesn't hurt to improve the code and make it more robust when on this :-) I did the same for `AbstractJWKParser` as well. Also made `ServerJWKParser` to not be abstract as I think it was `abstract` by accident and `abstract` caused this test to not be executed.

**Review by jonkoops — APPROVED** [review]

**Review by pskopek — APPROVED** [review]
LGTM!

**Comment by mposolda** [comment]
@pskopek @jonkoops Thanks for the review!

**Pull request merged**